From a0043c23d93dc9404d1117f4ad6e87e58848d311 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Fri, 26 Jan 2018 13:16:31 +0100
Subject: [PATCH 4/4] Fix redundant GETMETADATA requests when listing folders

When listing folders we call GETMETADATA for all folders to get their types,
then for every folder without type set we did another GETMETADATA call.
This is now fixed.
---
 lib/kolab_sync_backend.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/kolab_sync_backend.php b/lib/kolab_sync_backend.php
index 315a803..13a4e34 100644
--- a/lib/kolab_sync_backend.php
+++ b/lib/kolab_sync_backend.php
@@ -176,7 +176,7 @@ class kolab_sync_backend
             }
 
             // Activesync folder identifier (serverId)
-            $folder_type = $typedata[$folder];
+            $folder_type = $typedata[$folder] ?: 'mail';
             $folder_id   = self::folder_id($folder, $folder_type);
 
             $folders_list[$folder_id] = $this->folder_data($folder, $folder_type);
@@ -216,7 +216,7 @@ class kolab_sync_backend
                         array_pop($items);
 
                         $parent_name = implode($items, $delim);
-                        $parent_type = $typedata[$parent_name];
+                        $parent_type = $typedata[$parent_name] ?: 'mail';
                         $parent_id   = self::folder_id($parent_name, $parent_type);
 
                         if (isset($folders[$parent_id])) {
-- 
2.14.3

